addon/components/uni-country.js

```diff
@@ -0,0 +1,63 @@
+import Component from '@ember/component';
+import { computed } from '@ember/object';
+import { isBlank } from '@ember/utils';
+import { inject as service } from '@ember/service';
+import { A } from '@ember/array';
+import layout from '../templates/components/uni-country';
+import Countries from 'ember-cli-countries/enums/countries';
+
+export default Component.extend({
+  layout,
+  i18n: service(),
+
+  classNames: ['uni-country'],
+  classNameBindings: ['hasError:uni-country--error'],
+
+  countryCode: null,
+  countries: A(),
+  noResultsText: 'No results for ',
+  placeholderText: 'Type your country ',
+  pathToTranslateCountry: null,
+
+  onChange() {},
+
+  selectedCountry: computed('countries', 'countryCode', function() {
+    if (isBlank(this.get('countryCode'))) {
+      return '';
+    }
+
+    let country = this.get('countries').findBy('code', this.get('countryCode').toLowerCase());
+
+    return country ? country.name : '';
+  }),
+
+  init() {
+    this._super(...arguments);
+
+    this.set('countries', this.initializeCountries());
```

Comment:
Hmm... I'm very not very fond of this solution

Reply:
why is it not good? why is it better a computed property that depends on nothing?...

Reply:
It should depend on `i18n.locale`, so when you change the language, it's rerendered :)

Reply:
good idea